src/main/java/org/brackit/xquery/operator/TableJoin.java

```diff
@@ -57,21 +57,21 @@ private class TableJoinCursor implements Cursor {
     int itPos = 0;
     int itSize = 0;
 
-    public TableJoinCursor(Cursor lc, int lSize, int pad) {
-      this.lc = lc;
-      this.lSize = lSize;
+    public TableJoinCursor(Cursor cursor, int size, int pad) {
```

Comment:
It was short for left cursor I assume. However, the hash table should be built on the operator with the lower tuple size, so it can be either the cursor over the right or left operator tuples. I hope I got this right.